tools/test_containers_compose

```diff
@@ -20,9 +18,16 @@ wait_until() {
 }
 
 setup_containers() {
-    "$thisdir"/retry sudo docker compose build
+    if command -v retry > /dev/null; then
+        retry="retry -e -s 30 -r 7"
+    else
+        curl -sS https://raw.githubusercontent.com/okurz/retry/main/retry > retry
```

Comment:
I would prefer we didn't download scripts into the local checkout of the repository

Reply:
> I would prefer we didn't download scripts into the local checkout of the repository

what should we do instead?

Reply:
To be honest using "retry" instead of "retry" here also raises more questions. We also rely on this script for zypper issues in `build-docs` and `build_cache`.
So the download ideally happens either in a wrapper script or as part of Makefile.

We only have a "build" folder for docs I guess.

Reply:
> To be honest using "retry" instead of "retry" here also raises more questions.

Do you mean it's confusing if we don't rename the application because it happens to be called the same? Or do you confuse github.com/okurz/retry with tools/retry which are two different things?

> We also rely on this script for zypper issues in `build-docs` and `build_cache`. So the download ideally happens either in a wrapper script or as part of Makefile.

We rely on tools/retry so far and not github.com/okurz/retry . Maybe you want to suggest that we should use github.com/okurz/retry everywhere. That's actually what I would plan but didn't want to try everything in one go. So what do you suggest how should we proceed?

Reply:
> Do you mean it's confusing if we don't rename the application because it happens to be called the same? Or do you confuse github.com/okurz/retry with tools/retry which are two different things?

You are replacing the second with the first here. And we use tools/retry for the same use case so far. So if we use it here, we should probably use it everywhere?

Reply:
I mean in theory we could do that but it's not that easy as we need to ensure we have the application either downloaded or available from packages which is not the case on all OSs. And also tools/retry supports a stability test. Ok, I re-implemented the exponential sleep in a corresponding fashion now in tools/retry. Better?